system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp

```diff
@@ -1081,6 +1081,151 @@ bool AzureLogAnalyticsCurlClient::processSearchJsonResp(LogQueryResultDetails &
     return true;
 }
 
+bool AzureLogAnalyticsCurlClient::healthReport(StringBuffer & report, LogAccessHealthReportOptions options)
+{
+    try
+    {
+        report.appendf("\"ConnectionInfo\": { \"TargetALAWorkspaceID\": \"%s\" ", m_logAnalyticsWorkspaceID.str());
```

Comment:
For this and the other report building code, maybe it would be better to use the JSON builder functions in jstring like `appendJSONStringValue` and others. They can handle encoding and quoting in a standardized way.